software/base/src/main/java/org/apache/brooklyn/tasks/kubectl/JobBuilder.java

```diff
@@ -121,6 +130,11 @@ public String build(){
             containerSpec.setWorkingDir(workingDir);
         }
         containerSpec.setImage(imageName);
+        if(Strings.isNonBlank(imagePullPolicy)) {
+            if (PULL_POLICY_ALLOWED_VALUES.contains(imagePullPolicy)) {
+                containerSpec.setImagePullPolicy(imagePullPolicy);
+            } // else stick with default, do not fail
```

Comment:
surely it would be better to fail?

Reply:
@geomacy, the default policy that `kubectl` sets is 'Always'. And now, we simply allow to configure it.

Reply:
@iuliana, to remove that code comment `do not fail`?

Reply:
agree - i'd say either:

* always set the user-supplied value, let kubectl fail if it is invalid (good if they might add new valid args)
* or if we are going to try to duplicate enforcement, use an enum and specify that as the type of the config key (allows type introspection to determine valid values)

silently ignoring is going to confuse a user (esp if kubectl is case insensitive and we just drop things that don't match our arbitrary case!)

Reply:
@ahgittin `enum` it is then.

Reply:
> [ahgittin](https://github.com/ahgittin) [2 days ago](https://github.com/apache/brooklyn-server/pull/1329#discussion_r910735955)  silently ignoring is going to confuse a user (esp if kubectl is case insensitive and we just drop things that don't match our arbitrary case!)

That was my thought.

> [iuliana](https://github.com/iuliana) [2 days ago](https://github.com/apache/brooklyn-server/pull/1329#discussion_r911071862)
@geomacy Added enum for imagePullPolicy that validates the value, before the task is even created.

looks great thanks @iuliana